File: SofiForce.Web/ClientApp/src/app/core/Models/SearchModels/ClientComplainSearchModel.ts
import { BaseSearchModel } from "./baseSearchModel";

export interface ClientComplainSearchModel extends BaseSearchModel {
    complainCode: string;
    branchId: number | null;
    branchCode: string | '';

    representativeId: number | null;
    representativeCode: string | '';

    clientId: number | null;
    clientCode: string | '';

    complainDate: Date | null;
    complainTypeId: number | null;
    complainTypeDetailId: number | null;
    phone: string;
    priorityId: number | null;
    complainStatusId: number | null;
    isClosed: number | null;
    departmentId: number | null;
}

File: SofiForce.Models/Models/SearchModels/ClientComplainSearchModel.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class ClientComplainSearchModel : BaseSearchModel
    {
        public string ComplainCode { get; set; }
        public int? BranchId { get; set; }
        public int? RepresentativeId { get; set; }
        public int? ClientId { get; set; }
        public DateTime? ComplainDate { get; set; }
        public int? ComplainTypeId { get; set; }
        public int? ComplainTypeDetailId { get; set; }
        public string Phone { get; set; }
        public int? PriorityId { get; set; }
        public int? ComplainStatusId { get; set; }
        public int? IsClosed { get; set; }
        public int? DepartmentId { get; set; }

    }
}